Mathlib/Algebra/Algebra/NonUnitalHom.lean

```diff
@@ -45,32 +45,52 @@ non-unital, algebra, morphism
 
 set_option autoImplicit true
 
-
 universe u v w w₁ w₂ w₃
 
-variable (R : Type u) (A : Type v) (B : Type w) (C : Type w₁)
+variable {R : Type*} {S : Type*} -- [Monoid R] [Monoid S] (φ : R →* S)
+--   (A : Type v) (B : Type w) (C : Type w₁)
 
 /-- A morphism respecting addition, multiplication, and scalar multiplication. When these arise from
 algebra structures, this is the same as a not-necessarily-unital morphism of algebras. -/
-structure NonUnitalAlgHom [Monoid R] [NonUnitalNonAssocSemiring A] [DistribMulAction R A]
-  [NonUnitalNonAssocSemiring B] [DistribMulAction R B] extends A →+[R] B, A →ₙ* B
+structure NonUnitalAlgHom
+    [Monoid R] [Monoid S] (φ : R →* S)
+    (A : Type v) (B : Type w)
+    [NonUnitalNonAssocSemiring A] [DistribMulAction R A]
+    [NonUnitalNonAssocSemiring B] [DistribMulAction S B]
+    extends A →ₑ+[φ] B, A →ₙ* B
 #align non_unital_alg_hom NonUnitalAlgHom
 
 @[inherit_doc NonUnitalAlgHom]
 infixr:25 " →ₙₐ " => NonUnitalAlgHom _
 
 @[inherit_doc]
-notation:25 A " →ₙₐ[" R "] " B => NonUnitalAlgHom R A B
+notation:25 A " →ₙₐ[" R "] " B => NonUnitalAlgHom (MonoidHom.id R) A B
+
+@[inherit_doc]
+notation:25 A " →ₛₙₐ[" φ "] " B => NonUnitalAlgHom φ A B
 
 attribute [nolint docBlame] NonUnitalAlgHom.toMulHom
 
+/-- `NonUnitalAlgSemiHomClass F φ A B` asserts `F` is a type of bundled algebra homomorphisms
+from `A` to `B` which are equivariant with respect to `φ`.  -/
+class NonUnitalAlgSemiHomClass (F : Type*)
+    {R S : outParam (Type*)} [Monoid R] [Monoid S] (φ : outParam (R →* S))
+    (A : outParam (Type*)) (B : outParam (Type*))
+    [NonUnitalNonAssocSemiring A] [NonUnitalNonAssocSemiring B]
+    [DistribMulAction R A] [DistribMulAction S B]
+    extends DistribMulActionSemiHomClass F φ A B, MulHomClass F A B
+#align non_unital_alg_hom_class NonUnitalAlgSemiHomClass
+
+-- TODO (ACL) : why doesn't it work later?
```

Comment:
What do you mean precisely by "work later"?

Reply:
Probably a comment I didn't delete. I don't remember what I meant.